packages/compiler/src/myst.ts

```diff
@@ -36,5 +59,106 @@ export function parsePanel(contents: string): Array<MystCard> {
     parsedCard.body = body;
     splitCards.push(parsedCard);
   }
-  return splitCards;
+
+  let myPanel: MystPanel = { cards: splitCards };
+  if (panelStyle) {
+    myPanel = { ...myPanel, style: panelStyle };
+  }
+  return myPanel;
+}
+
+export function parseStyling(contents: string): [string, string] {
+  let yamlBlock = "";
+  let returnContents = "";
+  const contentLines = contents.split("\n");
+  if (contents.startsWith(":")) {
+    const yamlLines: Array<string> = [];
+    while (contentLines) {
+      if (!ltrim(contentLines[0]).startsWith(":")) {
+        break;
+      } else {
+        // TODO: Find way to fix non-null assertion warning
+        yamlLines.push(ltrim(contentLines.shift()!));
+      }
+    }
+    yamlBlock = yamlLines.join("\n");
+  }
+
+  returnContents = contentLines.join("\n").trim();
+  const returnPanel: [string, string] = [yamlBlock, returnContents];
+  return returnPanel;
+}
+
+export function parseYamlBlock(yamlBlock: string): Record<string, unknown> {
+  const styleContents = yamlBlock.split("\n");
+  const stylingRegex = /^:([a-z0-9]+):\s*([a-z\s\-0-9]+)/i;
+  // return object with key value pairs
+  let styles = {};
+  for (const line of styleContents) {
+    const stylingArray: RegExpExecArray | null = stylingRegex.exec(line);
+    if (stylingArray && stylingArray.length > 2) {
+      const key: string = stylingArray[1];
+      const value: string = stylingArray[2];
+      styles = { ...styles, [key]: value };
+    }
+  }
+  return styles;
+}
+
+function ltrim(rawString: string) {
+  return rawString.replace(/^\s+/gm, "");
+}
+
+export function mergeStyles(
+  panelStyle: MystStyling,
+  cardStyle: MystStyling
+): MystStyling {
+  let k: keyof MystStyling;
+  for (k in panelStyle) {
+    if (k in cardStyle) {
+      const panelPropDict = classesStringToKeyValues(panelStyle[k]!);
+      const cardPropDict = classesStringToKeyValues(cardStyle[k]!);
+      // merge Dicts
+      const mergedProps = { ...panelPropDict, ...cardPropDict };
+      cardStyle[k] = propDictToString(mergedProps);
+    } else {
+      cardStyle[k] = panelStyle[k];
+    }
+  }
+  return cardStyle;
+}
+
+export function classesStringToKeyValues(
+  classesString: string
+): Record<string, string> {
+  const htmlClasses = classesString.split(/\s+/);
+  let classDictionary = {};
+  for (const htmlClass of htmlClasses) {
+    const classArray = htmlClass.split("-");
+    let key, value;
+    // deal with single prop like "shadow"
+    if (classArray.length === 1) {
+      key = classArray[0];
+      value = "";
+    } else {
+      value = classArray.pop();
+      key = classArray.join("-");
+    }
```

Comment:
Could be expressed more concisely with array destructuring

```suggestion
    let [key, value] = [classArray[0], classArray.slice(1).join("-");
```

Reply:
Agreed, but needs to be slightly tweaked because the last value in the array is the key, (i.e. ```col-lg-6``` results in key ```col-lg```, value ```6```)

Reply:
Managed with https://github.com/irydium/irydium/pull/213/commits/f21a5e9b3281654363601087fb2650a2fff6f4d7